cylc/flow/subprocpool.py

```diff
@@ -308,6 +310,12 @@ def _poll_proc_pipes(self, proc, ctx):
                     if ctx.err is None:
                         ctx.err = ''
                     ctx.err += data
+            if received_data and not all(received_data):
+                # if no data was pushed down the pipe exit the polling loop
+                # (this suppresses an infinite polling-loop observed
+                # on darwin) see
+                # https://github.com/cylc/cylc-flow/issues/3535
+                return
```

Comment:
In my environment execution hits this line, usually every minute or two.

Reply:
But is there no negative consequence of this, to the scheduler's operation?

Reply:
I don't think so, the purpose of this polling is to read form stdout/err to unblock these streams incase they got too big I think. This polling is run for each running process every main loop. So if we exit this polling loop early it will get re-run the next time the `subprocpool.process` is called (which is every main loop).

Note this polling gets skipped completely on platforms where `select.poll` isn't available (note whilst MacOS has POSIX compatibile `poll` the system Python [at least on some builds] ships without `select.poll` for some strange reason, perhaps hence we haven't seen this before [I'm using Conda Python])

Reply:
OK, can you add this explanation as a comment about this to the code, so we know what's going on in future?

Reply:
Added a bit more info and a link to this PR.